File: Conceitos de LPs/Tipos de Dados/Conjunto/Conjunto.java
//Java
import java.util.HashSet;
import java.util.Set;

public class Conjunto {
    public static void main(String[] args) {
        // Definindo e inicializando conjuntos
        Set<Integer> conjuntoA = new HashSet<>();
        Set<Integer> conjuntoB = new HashSet<>();

        for (int i = 1; i <= 5; i++) conjuntoA.add(i);
        for (int i = 4; i <= 8; i++) conjuntoB.add(i);

        // União
        Set<Integer> uniao = new HashSet<>(conjuntoA);
        uniao.addAll(conjuntoB);

        // Intersecção
        Set<Integer> intersecao = new HashSet<>(conjuntoA);
        intersecao.retainAll(conjuntoB);

        // Diferença
        Set<Integer> diferenca = new HashSet<>(conjuntoA);
        diferenca.removeAll(conjuntoB);

        // Exibindo resultados
        System.out.println("União: " + uniao);
        System.out.println("Intersecção: " + intersecao);
        System.out.println("Diferença: " + diferenca);
    }
}

File: Conceitos de LPs/Tipos de Dados/Conjunto/Conjunto.pas
//Pascal
program Conjunto;
uses SysUtils;

type
  TConjunto = set of 1..10;

var
  conjuntoA, conjuntoB, uniao, intersecao, diferenca: TConjunto;
  i: Integer;

begin
  // Definindo e inicializando conjuntos
  conjuntoA := [1, 2, 3, 4, 5];
  conjuntoB := [4, 5, 6, 7, 8];

  // União
  uniao := conjuntoA + conjuntoB;

  // Intersecção
  intersecao := conjuntoA * conjuntoB;

  // Diferença
  diferenca := conjuntoA - conjuntoB;

  // Exibindo resultados
  Write('União: ');
  for i := 1 to 10 do
    if i in uniao then Write(i, ' ');
  Writeln;

  Write('Intersecção: ');
  for i := 1 to 10 do
    if i in intersecao then Write(i, ' ');
  Writeln;

  Write('Diferença: ');
  for i := 1 to 10 do
    if i in diferenca then Write(i, ' ');
  Writeln;
end.
